src/usr/local/www/system.php

```diff
@@ -242,6 +242,22 @@ function is_timezone($elt) {
 		// Put the user-entered list back into place so it will be redisplayed for correction.
 		$pconfig['dnsserver'] = $dnslist;
 	} else {
+		// input validation passed, so we can proceed with removing static routes for dead DNS gateways
+		if (is_array($config['system']['dnsserver'])) {
+		  	$dns_servers_arr = $config['system']['dnsserver'];
+	 		foreach ($dns_servers_arr as $arr_index => $this_dnsserver) {
+	   			$i = (int)$arr_index + 1;
+	   			$this_dnsgw = $config['system']['dns'.$i.'gw'];
+				unset($gatewayip);
+				unset($inet6);
+				if ((!empty($this_dnsgw)) && ($this_dnsgw != 'none') && (!empty($this_dnsserver))) {
+					$gatewayip = lookup_gateway_ip_by_name($this_dnsgw);
+					$inet6 = is_ipaddrv6($gatewayip) ? '-inet6 ' : '';
+					mwexec("/sbin/route -q delete -host {$inet6}{$this_dnsserver} {$gatewayip}");
```

Comment:
You do not need to specify the gateway when deleting a route, and doing so could be more error prone than leaving it off in this case.

Reply:
Got it. Fixed.